stacker/plan.py

```diff
@@ -210,65 +220,40 @@ def _single_run(self):
                 )
                 continue
 
-            # Kick off watchers - used for tailing the stack
-            if (
-                not step.done and
-                self._watch_func and
-                step_name not in self._watchers
-            ):
-                process = multiprocessing.Process(
-                    target=self._watch_func,
-                    args=(step.stack,)
-                )
-                self._watchers[step_name] = process
-                process.start()
-
-            try:
-                status = step.run()
-            except CancelExecution:
-                status = SkippedStatus(reason="canceled execution")
-
-            if not isinstance(status, Status):
-                raise ValueError(
-                    "Step run_func must return a valid Status object. "
-                    "(Returned type: %s)" % (type(status)))
-            step.set_status(status)
+            if not step.submitted:
+                try:
+                    status = step.run()
+                except CancelExecution:
+                    status = SkippedStatus(reason="canceled execution")
 
-            # Terminate any watchers when step completes
-            if step.done and step_name in self._watchers:
-                self._terminate_watcher(self._watchers[step_name])
+                if not isinstance(status, Status):
+                    raise ValueError(
+                        "Step run_func must return a valid Status object. "
+                        "(Returned type: %s)" % (type(status)))
+                step.set_status(status)
+            else:
+                self.poll()
 
         return self.completed
 
-    def _terminate_watcher(self, watcher):
-        if watcher.is_alive():
-            watcher.terminate()
-            watcher.join()
-
     def execute(self):
         """Execute the plan.
 
         This will run through all of the steps registered with the plan and
         submit them in parallel based on their dependencies.
         """
-
         attempts = 0
         last_md5 = self.md5
-        try:
-            while not self.completed:
-                if (
-                    not attempts % self.check_point_interval or
-                    self.md5 != last_md5
-                ):
-                    last_md5 = self.md5
-                    self._check_point()
-
-                attempts += 1
-                if not self._single_run():
-                    self._wait_func(self.sleep_time)
-        finally:
-            for watcher in self._watchers.values():
-                self._terminate_watcher(watcher)
+
+        while not self.completed:
+            if (not attempts % self.check_point_interval or
+                    self.md5 != last_md5):
+                last_md5 = self.md5
+                self._check_point()
+
+            attempts += 1
+            if not self._single_run():
+                self._wait_func(self.sleep_time)
```

Comment:
Do we need the wait_func here anymore?  Since `_single_run` long polls, I think this is just going to slow things down unnecessarily.